asyncio_mqtt/client.py

```diff
@@ -104,7 +104,25 @@ def __init__(
         self.ciphers = ciphers
         self.keyfile_password = keyfile_password 
         
-
+# Proxy parameters class 
+class ProxySettings:
+    def __init__(
+        self,
+        *, 
+        proxy_type: int = None,
```

Comment:
You forgot the `Optional` part here. :)

Reply:
If you look at the description here: https://github.com/eclipse/paho.mqtt.python/blob/225ab3757f6818ba85eb80564948d1c787190cba/src/paho/mqtt/client.py#L875

It says 
```
(Required)
proxy_type: One of {socks.HTTP, socks.SOCKS4, or socks.SOCKS5}
proxy_addr: IP address or DNS name of proxy server
(Optional)
proxy_rdns: boolean indicating whether proxy lookup should be performed
    remotely (True, default) or locally (False)
proxy_username: username for SOCKS5 proxy, or userid for SOCKS4 proxy
proxy_password: password for SOCKS5 proxy
```

In line with that shouldnt it be like this, or should I change it to Optional?

Reply:
Ah, good point. :+1:  If those arguments are _required_ then let's not use default valures (`= None`) for them.
Looking at the [paho implementation](https://github.com/eclipse/paho.mqtt.python/blob/225ab3757f6818ba85eb80564948d1c787190cba/src/paho/mqtt/client.py#L3477), we could have a signature like this:

```python
def __init__(
    self,
    *, 
    proxy_type: int,
    proxy_addr: str,
    proxy_rdns: Optional[bool] = True,
    proxy_username: Optional[str] = None,
    proxy_password: Optional[str] = None
): ...
```

This way, the user must supply the `proxy_type` and `proxy_addr` arguments.

> Also do you think it is a good idea to add some documentation so that the users know that the functionality exists (e.g a small example) in the README.md in another PR?

Yes, that's a good idea. :+1: 
Specifically, we should let people know that this whole proxy is an ["extra" feature (even in paho) that requires the `PySocks` dependency](https://github.com/eclipse/paho.mqtt.python/blob/master/setup.py#L18).

Reply:
I have fixed it and committed it as well. 
Alright. I will work on the documentation and create a PR for that